src/test/java/com/codeus/winter/config/ClassPathBeanDefinitionScannerTest.java

```diff
@@ -0,0 +1,41 @@
+package com.codeus.winter.config;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.codeus.winter.config.impl.BeanDefinitionRegistryImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ClassPathBeanDefinitionScannerTest {
+    private BeanDefinitionRegistry registry;
+    private ClassPathBeanDefinitionScanner scanner;
+
+    @BeforeEach
+    void setUp() {
+        registry = new BeanDefinitionRegistryImpl();
+        scanner = new ClassPathBeanDefinitionScanner(registry);
+    }
+
+    @Test
+    void testScanPackagesWhenValidBasePackagesShouldReturnCorrectCount() {
```

Comment:
In addition to this test case, please add one more: scan a package with 2 classes and assert that the registered bean definition count == 2, and after that using the same registry instance, scan another package with 1 class and assert that the registered bean definition count == 1 (see another comment about achieving this behavior).

Reply:
removed test class